mat64/io.go

```diff
@@ -60,6 +62,42 @@ func (m Dense) MarshalBinary() ([]byte, error) {
 	return buf, nil
 }
 
+// MarshalBinaryTo encodes the receiver into a binary form and writes it into w.
+// MarshalBinaryTo returns the number of bytes written into w and an error, if any.
+//
+// See MarshalBinary for the on-disk layout.
+func (m Dense) MarshalBinaryTo(w io.Writer) (int, error) {
+	// please keep Dense.MarshalBinaryTo in sync with Dense.MarshalBinary
```

Comment:
Don't make a comment, just add a test that checks that the []byte-based and streaming methods give the same result - marshal to a bytes buffer and to a [] and round-trip that to a dense/vector via a bytes reader and the []byte.

All locations below as well.